### PR TITLE
Use active theme colors for styled text in the debug console

### LINKED ISSUE #42388: Debug Console should use theme-defined red/green colors
Themes define custom ansi colors, for example in Monokai, 

```json
"terminal.ansiBlack": "#333333",
"terminal.ansiRed": "#C4265E",
"terminal.ansiGreen": "#86B42B",
"terminal.ansiYellow": "#B3B42B",
```

The terminal uses these, but the debug console uses the same colors for every theme. Would be cool if it can use the theme colors too.

Example in Monokai

Terminal
image

Console
image

### LINKED ISSUE #63986: Support high intensity ansi colours in debug console
This issue is basically to revive one I previously found: Microsoft/vscode#21423

The issue was closed when underlining functionality was added, but the original request, for high intensity colours, was never addressed. Any chance these could be added?

I was using the c++ extension and my program would log using the ANSI codes. I saw no visible difference in colour between, for example:

Red (\033[31m)
Bright Red (\033[31;1m)
aixterm Bright Red (\033[91m)

Colour code output

Now clearly you've gone for the "bold" interpretation of the "bold or high intensity" option of having ";1" after your colour. But is there any chance of the aixterm versions working?

### PR DESCRIPTION
**Pull request by iansan5653**
~Draft because I'm not able to run tests / compile on this machine at the moment, but the code is pretty much done and ready for input.~

A problem with this approach is that after text is logged, if the user changes the theme, the logged text won't change color. One thing I considered was keeping the CSS rules but with CSS variables and dynamically changing the variables, but that seemed like it was introducing a lot of unnecessary complexity for an issue that isn't even much of an issue at all (the debug console is cleared often anyway).

Will fix #63986 and fix #42388.

### CODE REVIEW

**Comment by isidorn** [comment]
@iansan5653 ok, let me know when you ran all the tests and they are good.
Did you consider listening onThemeChange and just refreshing the whole repl tree, so the colors get updated. Basically `onThemeChange` just schedule a [`refreshScheduler`](https://github.com/Microsoft/vscode/blob/5ef97e39c618cf4d697fb1c3358d1646f764d1ac/src/vs/workbench/contrib/debug/browser/repl.ts#L344)

**Comment by iansan5653** [comment]
@isidorn I've gotten everything working, except the tests are failing for apparently no reason. I think it's because the `IThemeService` instantiated in the test file isn't working. I'm using:
```
themeService = new TestThemeService();
```
Is this the correct way to do it for tests?

**Comment by isidorn** [comment]
@iansan5653 that's the proper way to create a `TestThemeService`.
I can see the more precise errors here https://github.com/Microsoft/vscode/pull/71458/checks?check_run_id=104419801
Check them out.
Maybe you need to improve the TestThemeService, not sure where it exactly fails.
Debug the tests, we even have a launch configuration that runs the unit tests.

Screenshot 2019-04-18 at 10 33 52

**Comment by iansan5653** [comment]
Yeah, I've been trying to fix those for a while now. I know the class is being applied because when I compile and run it, I can see the class in the developer tools. I just don't know why the assertion is failing. I'll keep trying to resolve.

File: src/vs/workbench/contrib/debug/browser/debugANSIHandling.ts [hunk 1]
```diff
@@ -199,7 +213,43 @@ export function handleANSIOutput(text: string, linkDetector: LinkDetector): HTML
...
+			colorIndex = styleCode - 40;
+			colorType = 'background';
+		} else if (styleCode >= 100 && styleCode <= 107) {
+			colorIndex = (styleCode - 100) + 8; // High-intensity (bright)
+			colorType = 'background';
+		}
+
+		if (colorIndex !== undefined && colorType) {
+			const colorName = ansiColorIdentifiers[colorIndex];
+			const color = theme.getColor(colorName);
```
Comment (iansan5653) [inline src/vs/workbench/contrib/debug/browser/debugANSIHandling.ts]: @isidorn I was able to narrow down the issue to this line. At this point, `colorName` is `terminal.ansiBlack` but `theme.getColor(colorName)` returns `undefined`. This only happens in the test environment; when running in VSCode, the correct color value is returned, so I'm pretty sure it's probably the test theme service initialization. I'm not really sure what else I can try at this point.

Comment (isidorn) [reply src/vs/workbench/contrib/debug/browser/debugANSIHandling.ts]: You need to improve the getColor method of the TestThemeService just so you get the tests to pass. You need to return some mock color
Try it out and if it turns out too complex I can look into it

Comment (iansan5653) [reply src/vs/workbench/contrib/debug/browser/debugANSIHandling.ts]: Thanks for pointing that out! I ended up instantiating a custom `TestTheme` with the necessary colors; hope that works alright.

**Comment by isidorn** [comment]
Thanks a lot. However we are currently in endgame before the release which means we only accept a limited set of changes. Due to that assigning this to May when I plan to merge it (in about 10 days)

**Comment by isidorn** [comment]
Thanks a lot for this PR. And sorry for merging with a bit of a delay.

**Comment by iansan5653** [comment]
No problem! Thanks for the advice and the merge.

**Pull request merged**